src/util.ts

```diff
@@ -201,12 +201,35 @@ export function hasTrailingSlash(path: string): boolean {
 }
 
 export function isExtensionless(path: string): boolean {
-  // `path.lastIndexOf('.') == -1` is a Node-specific
-  //  heuristic here. There aren't any files that don't
-  //  have file extensions, so, if there are no file extensions
-  //  specified in the url, treat it like a directory.
-  //  One exception is `latest-vXX.x`, which is a directory
-  return path.lastIndexOf('.') === -1 || path.toLowerCase().endsWith('.x');
+  // `path.lastIndexOf('.') == -1` is a Node-specific heuristic here.
+  //  There aren't any files in the bucket that don't have file extensions,
+  //  so, if there is no file extension specified in the url, treat it
+  //  like a directory.
+  //
+  // Two exceptions:
+  //  - `latest-vXX.x` directories
+  //  - `vX.X.X` directories
+
+  const extensionDelimiter = path.lastIndexOf('.');
+  if (extensionDelimiter === -1) {
+    return true;
+  }
+
+  const fileExtension = path.substring(extensionDelimiter + 1); // +1 to remove the `.`
+
+  // `latest-vXX.x` directory
+  if (fileExtension.toLowerCase() === 'x') {
```

Comment:
What if we check if it is a number or a string with more than two digits? I do not know of any valid file extension of just one digit, right?

Reply:
I think it should support any number of characters just because there are folders such as https://nodejs.org/download/release/v0.1.100/

Reply:
No what I meant is the following:

- Instead of checking against the file extension being "X" you check if it is one alphanumeric character or N digits number. Like it can be a number of any length but a string/character of 1 length.

It would even make sense to test with regex.

Reply:
Ahh I see, yeah I think that'd be a better way of doing this